public/js/auth.js

```
@@ -167,3 +167,364 @@ async function verifyUserBeforeAction(actionDescription = 'perform this action')
+                currentUser = user;
+
+                await loadUserProfile();
+
+                showAuthenticatedUI();
+
+                const landingPage = document.getElementById('landingPage');
+                const appContainer = document.getElementById('app');
+                if (landingPage) landingPage.style.display = 'none';
+                if (appContainer) appContainer.style.display = 'flex';
+
+                if (!userProfile || !userProfile.username) {
```

ESLint

[error] 'userProfile' is not defined.

(no-undef)